libnmstate/nm/active_connection.py

```diff
@@ -205,3 +197,27 @@ def _is_device_master_type(nmdev):
         )
         return is_master_type
     return False
+
+
+def _ip_config_can_be_treat_as_activated(nm_dev, nm_ac):
```

Comment:
This functions does too many things IMO and its name is not describing what it does, only how the caller interprets its output.

It will be nicer to read and change in the future if we can just read the full condition at the caller functions:
- If device reached ip config state AND
- this is a master device OR is_ip_dynamic(4) OR is_ip_dynamic(6) OR is_ip_dynamic(4,6)

is_ip_dynamic returns true only if both stacks have dynamic or if one has it and the other is READY with the other.

Reply:
Condition check has been moved back the the caller.